static/js/components/LearnerSearch.js

```diff
@@ -41,10 +42,28 @@ const makeSearchkitTranslations: () => Object = () => {
       translations[stateCode] = iso3166.data[code].sub[stateCode].name;
     }
   }
-
   return translations;
 };
 
+const SEMESTER_SEASON_NUMBER_MAP = {
+  'Spring': '1',
+  'Summer': '2',
+  'Fall': '3'
+};
+
+const sortSemesterBuckets = (buckets: Array): Array => {
+    if (buckets.length > 0) {
+      return _.sortBy(buckets, [
+        (bucket) => {
+          let yearSeasonArray = bucket.key.split(/[^\w]+/);
+          return yearSeasonArray[0] + SEMESTER_SEASON_NUMBER_MAP[yearSeasonArray[0]];
+        }
+      ]).reverse();
+    } else {
+      return buckets;
+    }
+};
```

Comment:
Is there a reason why you need the `buckets.length > 0` control flow? my suggestion would be something like this:

```js
const sortSemesterBuckets = R.compose(R.reverse, R.sortBy(R.compose(
  str => `${str}${SEMESTER_SEASON_NUMBER_MAP[str]}`,
  R.head,
  R.split(/[^\w]+/),
  R.prop('key')
)));
```

which might make it a little clearer.